### PR TITLE
server: disable security on /api/v2/ when running insecure

### PR DESCRIPTION
**Pull request by dhartunian**
Previously, the session validation log in `/api/v2/` would still run even when
the cluster was running in "insecure" mode. This made it cumbersome to test new
features while developing, and also caused new challenges when developing DB
Console features that use `/api/v2/` endpoints.

Now, when the cluster has the insecure flag set to true, web session check
failures won't cause failures on endpoints and the session username will be set
to "root" automatically in the context.

Release note (security update): HTTP API endpoints under the `/api/v2/` prefix,
will allow requests through when the cluster is running in "insecure" mode.
When the cluster is running in "insecure" mode requests to these endpoints will
have the username set to "root".

Release justification: low-risk high-benefit change to existing functionality.

### CODE REVIEW

File: pkg/server/api_v2_auth.go [hunk 1]
```diff
@@ -335,36 +336,40 @@ func (a *authenticationV2Mux) getSession(
...
 	}
 	if !valid {
 		err := errors.New("the provided authentication session could not be validated")
-		http.Error(w, err.Error(), http.StatusUnauthorized)
-		return "", nil, err
+		return "", nil, http.StatusUnauthorized, err
 	}
 
-	return username, sessionCookie, nil
+	return username, sessionCookie, 200, nil
```
Comment (knz) [inline pkg/server/api_v2_auth.go:351]: http.StatusOK?

**Review by knz — APPROVED** [review]
thanks

**Review by dhartunian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/86417)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @xinhaoz)

___
*[`pkg/server/api_v2_auth.go` line 351 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/86417#-N9mZkrm26gSdtIrF6P1-r1-351:-N9moEbfBQBjVsy9yGDD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c9e7d1dbae6fa7ca088d4a3a995e88fe9a6560b8/pkg/server/api_v2_auth.go#L351)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

http.StatusOK?
</blockquote></details>

Done.

**Comment by dhartunian** [comment]
TFTR

bors r=knz

**Pull request merged**